src/astx/context_manager.py

```diff
@@ -0,0 +1,117 @@
+"""ASTx class for With Statement (Context Manager)."""
+
+from __future__ import annotations
+
+from typing import Dict, List, Optional, cast
+
+from public import public
+
+from astx.base import (
+    NO_SOURCE_LOCATION,
+    ASTKind,
+    ASTNodes,
+    DataTypesStruct,
+    Expr,
+    Identifier,
+    ReprStruct,
+    SourceLocation,
+    StatementType,
+)
+from astx.blocks import Block
+
+
+@public
+class WithItem:
+    """AST class representing an item inside a `with` statement."""
+
+    def __init__(
+        self, context_expr: Expr, instance_name: Optional[Identifier] = None
+    ) -> None:
+        """Initialize a WithItem instance.
+
+        Args:
+            context_expr: The expression representing the context manager.
+            instance_name: Optional variable name to bind the context manager.
+        """
+        self.context_expr = context_expr
+        self.instance_name = instance_name
+
+    def __str__(self) -> str:
+        """Return string representation of the WithItem."""
+        if self.instance_name:
+            return f"{self.context_expr} as {self.instance_name}"
+        return str(self.context_expr)
+
+    def _prepare_struct(
+        self, key: str, value: DataTypesStruct, simplified: bool = False
+    ) -> Dict[str, DataTypesStruct]:
+        """Prepare structural representation.
+
+        Returns
```

Comment:
There is no need to add a Returns section.

fix the other functions as well. just add the title of the docstring
In the feature we will have some special way to describe each item in the docstring arguments on top of the annotation type